upmed-api/src/api/patient/patient.py

```diff
@@ -1,7 +1,294 @@
+from ....src.util.util import Auth
+from ....src.models.patient import Patient
+from ....src.models.health_event import HealthEvent
+
 
 patient_endpoints = Blueprint('patient', __name__)
+pdb = Database()
+pat = pdb.getPatients()
+auth = Auth()
+
 
-@patient_endpoints.route('/')
+@patient_endpoints.route('/', methods=['POST'])
 def root():
-    return "patient"
+    return "GO TO AN ENDPOINT"
+
+
+@patient_endpoints.route('/logIn', methods=['POST'])
+def login():
+    try:
+        pid = request.json['id']
+        email = request.json['email']
+        res = pat.document(str(pid)).get()
+        res = res.to_dict()
+        # print(pid)
+        if res['email'] == email:
+            utype = "PATIENT"
+            auth_token = auth.encode_auth_token(pid, utype)
+            resp = {
+                "googleId": pid,
+                "token": auth_token.decode()
+            }
+            # resp = res
+            # print(resp)
+            return jsonify(resp), 200
+        else:
+            return "False", 404
+    except Exception as e:
+        return f"An Error Occured: {e}"
+
+
+@patient_endpoints.route('/signUp', methods=['POST'])
+def signup():
+    post_data = request.get_json()
+    try:
+        patient = Patient(
+            id= post_data.get('id'),
+            firstName = post_data.get('firstName'),
+            lastName = post_data.get('lastName'),
+            phone = post_data.get('phone'),
+            email = post_data.get('email'),
+            dateOfBirth = post_data.get('dateOfBirth'),
+            sex = post_data.get('sex'),
+            profilePicture = '',
+            height = post_data.get('height'),
+            weight = post_data.get('weight'),
+            drinker = post_data.get('drinker'),
+            smoker = post_data.get('smoker'),
+            calendar=[],
+            health=[],
+            doctors=[]
+        )
+        try:
+            patient.profilePicture = post_data.get('profilePicture')
+        except KeyError:
+            patient.profilePicture = 'https://www.flaticon.com/svg/static/icons/svg/147/147144.svg'
+
+        utype = "PATIENT"
+        pat.document(patient.id).set({
+            "id": patient.id,
+            "firstName": patient.firstName,
+            "lastName": patient.lastName,
+            "phone": patient.phone,
+            "email": patient.email,
+            "dateOfBirth": patient.dateOfBirth,
+            "sex": patient.sex,
+            "profilePicture": patient.profilePicture,
+            "height": patient.height,
+            "weight": patient.weight,
+            "drinker": patient.drinker,
+            "smoker": patient.smoker,
+            "calendar": patient.calendar,
+            "health": patient.health,
+            "doctors": patient.doctors
+
+        })
+        auth_token = auth.encode_auth_token(patient.id, utype)
+        responseObject = {
+            'id': patient.id,
+            'auth_token': auth_token.decode()
```

Comment:
same as other comment about token

Reply:
That is to decode from the byte encoded string to a normal string @kenkenchuen

Reply:
we want the encoded string to pass around, front end will decode it itself